### PR TITLE
Fix & improve new proposal flow

### PR DESCRIPTION
**Pull request by isocolsky**

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -1535,7 +1539,8 @@ WalletService.prototype.createTxLegacy = function(opts, cb) {
  * @param {string} opts.outputs[].message - A message to attach to this output.
  * @param {string} opts.message - A message to attach to this transaction.
  * @param {Array} opts.inputs - Optional. Inputs for this TX
- * @param {string} opts.feePerKb - Optional. Use an alternative fee per KB for this TX
+ * @param {string} opts.fee - Optional. Use an alternative fee for this TX (mutually exclusive with feePerKb)
```
Comment (matiu) [inline lib/server.js]: nice!

File: lib/common/defaults.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ var Defaults = {};
 Defaults.DEFAULT_FEE_PER_KB = 10000;
 Defaults.MIN_FEE_PER_KB = 0;
 Defaults.MAX_FEE_PER_KB = 1000000;
-Defaults.MAX_TX_FEE = 1 * 1e8;
+Defaults.MIN_TX_FEE = 0;
+Defaults.MAX_TX_FEE = 0.1 * 1e8;
```
Comment (matiu) [inline lib/common/defaults.js]: hard limits looks fine. I thought about suggesting increasing MIN_TX_FEE, but I guess that for the general case 0 is OK

File: lib/server.js [hunk 3]
```diff
@@ -1648,13 +1659,11 @@ WalletService.prototype.publishTx = function(opts, cb) {
...
-        // Verify signingKey
-        if (opts.proposalSignaturePubKey) {
-          if (opts.proposalSignaturePubKey != signingKey ||
-            !self._verifyRequestPubKey(opts.proposalSignaturePubKey, opts.proposalSignaturePubKeySig, copayer.xPubKey)
-          ) {
-            return cb(new ClientError('Invalid proposal signing key'));
-          }
+        // Save signature info for other copayers to check
+        txp.proposalSignature = opts.proposalSignature;
+        if (signingKey.selfSigned) {
```
Comment (matiu) [inline lib/server.js]: Good catch

File: test/integration/server.js [hunk 4]
```diff
@@ -2721,15 +2722,53 @@ describe('Wallet service', function() {
         });
       });
 
+      it('should be able to specify the final fee', function(done) {
```
Comment (matiu) [inline test/integration/server.js]: related:  https://github.com/bitpay/copay/issues/3111

File: test/integration/server.js [hunk 5]
```diff
@@ -2785,31 +2826,25 @@ describe('Wallet service', function() {
...
-            var proposalSignature = helpers.signMessage(raw, TestData.copayers[0].privKey_1H_0);
-            var pubKey = new Bitcore.PrivateKey(TestData.copayers[0].privKey_1H_0).toPublicKey().toString();
-            var pubKeySig = helpers.signMessage(pubKey, TestData.copayers[1].privKey_1H_0);
-
             var publishOpts = {
               txProposalId: txp.id,
-              proposalSignature: proposalSignature,
-              proposalSignaturePubKey: pubKey,
-              proposalSignaturePubKeySig: pubKeySig,
+              proposalSignature: helpers.signMessage(txp.getRawTx(), TestData.copayers[1].privKey_1H_0),
```
Comment (matiu) [inline test/integration/server.js]: :clap: nice that we dont have to pass proposalSignaturePub*

File: lib/common/defaults.js (hunk 2 above)
Comment (isocolsky) [inline lib/common/defaults.js]: I thought about this too. I think BWS should not exclude free transactions. Maybe in the future we can set a dynamic minimum based on the min_realy_fee and providing a flag to override the min?

File: test/integration/server.js (hunk 5 above)
Comment (isocolsky) [inline test/integration/server.js]: Yes I forgot those are already validated in the `addAccess` method.

**Pull request merged**